node/pkg/adminrpc/adminserver.go

```diff
@@ -891,48 +837,64 @@ func (s *nodePrivilegedService) SignExistingVAA(ctx context.Context, req *nodev1
 }
 
 func (s *nodePrivilegedService) DumpRPCs(ctx context.Context, req *nodev1.DumpRPCsRequest) (*nodev1.DumpRPCsResponse, error) {
-	rpcMap := make(map[string]string)
-
-	rpcMap["acalaRPC"] = *acalaRPC
-	rpcMap["algorandIndexerRPC"] = *algorandIndexerRPC
-	rpcMap["algorandAlgodRPC"] = *algorandAlgodRPC
-	rpcMap["aptosRPC"] = *aptosRPC
-	rpcMap["arbitrumRPC"] = *arbitrumRPC
-	rpcMap["auroraRPC"] = *auroraRPC
-	rpcMap["avalancheRPC"] = *avalancheRPC
-	rpcMap["baseRPC"] = *baseRPC
-	rpcMap["bscRPC"] = *bscRPC
-	rpcMap["celoRPC"] = *celoRPC
-	rpcMap["ethRPC"] = *ethRPC
-	rpcMap["fantomRPC"] = *fantomRPC
-	rpcMap["ibcLCD"] = *ibcLCD
-	rpcMap["ibcWS"] = *ibcWS
-	rpcMap["karuraRPC"] = *karuraRPC
-	rpcMap["klaytnRPC"] = *klaytnRPC
-	rpcMap["moonbeamRPC"] = *moonbeamRPC
-	rpcMap["nearRPC"] = *nearRPC
-	rpcMap["neonRPC"] = *neonRPC
-	rpcMap["oasisRPC"] = *oasisRPC
-	rpcMap["optimismRPC"] = *optimismRPC
-	rpcMap["polygonRPC"] = *polygonRPC
-	rpcMap["pythnetRPC"] = *pythnetRPC
-	rpcMap["pythnetWS"] = *pythnetWS
-	rpcMap["sei"] = "IBC"
-	if s.testnetMode {
-		rpcMap["sepoliaRPC"] = *sepoliaRPC
-	}
-	rpcMap["solanaRPC"] = *solanaRPC
-	rpcMap["suiRPC"] = *suiRPC
-	rpcMap["terraWS"] = *terraWS
-	rpcMap["terraLCD"] = *terraLCD
-	rpcMap["terra2WS"] = *terra2WS
-	rpcMap["terra2LCD"] = *terra2LCD
-	rpcMap["wormchainWS"] = *wormchainWS
-	rpcMap["wormchainLCD"] = *wormchainLCD
-	rpcMap["xplaWS"] = *xplaWS
-	rpcMap["xplaLCD"] = *xplaLCD
-
 	return &nodev1.DumpRPCsResponse{
-		Response: rpcMap,
+		Response: s.rpcMap,
 	}, nil
 }
+
+func VerifyReq(req *nodev1.InjectGovernanceVAARequest) {
```

Comment:
Could this function also be used in InjectGovernanceVAA() so that this big switch statement only occurs in one place?

Reply:
I think this whole code could use some good refactoring, which is largely out of scope of this PR (just trying to move this to a new package), but in this case I think it's a worthwile change to include to I made it as you suggested.

Reply:
Thank you for making this change.